lib/modules/versioning/devbox/index.spec.ts

```diff
@@ -0,0 +1,78 @@
+import devbox from '.';
+
+describe('modules/versioning/devbox/index', () => {
+  it.each`
+    version           | expected
+    ${'1'}            | ${false}
+    ${'01'}           | ${false}
+    ${'1.01'}         | ${false}
+    ${'1.1'}          | ${false}
+    ${'1.3.0'}        | ${true}
+    ${'2.1.20'}       | ${true}
+    ${'v1.4'}         | ${false}
+    ${'V0.5'}         | ${false}
+    ${'3.5.0'}        | ${true}
+    ${'4.2.21.Final'} | ${false}
+    ${'1234'}         | ${false}
+    ${'foo'}          | ${false}
+    ${'latest'}       | ${false}
+    ${''}             | ${false}
+    ${'3.5.0-beta.3'} | ${false}
+    ${'*'}            | ${false}
+    ${'x'}            | ${false}
+    ${'X'}            | ${false}
+    ${'~1.2.3'}       | ${false}
+    ${'>1.2.3'}       | ${false}
+    ${'^1.2.3'}       | ${false}
+    ${'1.2.3-foo'}    | ${false}
+    ${'1.2.3foo'}     | ${false}
+  `('isVersion("$version") === $expected', ({ version, expected }) => {
+    expect(!!devbox.isVersion(version)).toBe(expected);
+  });
+
+  it.each`
+    version           | isValid
+    ${'1'}            | ${true}
+    ${'01'}           | ${false}
+    ${'1.01'}         | ${false}
+    ${'1.1'}          | ${true}
+    ${'1.3.0'}        | ${true}
+    ${'2.1.20'}       | ${true}
+    ${'v1.4'}         | ${false}
+    ${'V0.5'}         | ${false}
+    ${'3.5.0'}        | ${true}
+    ${'4.2.21.Final'} | ${false}
+    ${'1234'}         | ${true}
+    ${'foo'}          | ${false}
+    ${'latest'}       | ${true}
+    ${''}             | ${false}
+    ${'3.5.0-beta.3'} | ${false}
+    ${'*'}            | ${false}
+    ${'x'}            | ${false}
+    ${'X'}            | ${false}
+    ${'~1.2.3'}       | ${false}
+    ${'>1.2.3'}       | ${false}
+    ${'^1.2.3'}       | ${false}
+    ${'1.2.3-foo'}    | ${false}
+    ${'1.2.3foo'}     | ${false}
+  `('isValid("$version") === $isValid', ({ version, isValid }) => {
+    expect(!!devbox.isValid(version)).toBe(isValid);
+  });
+
+  it.each`
+    version    | range       | expected
+    ${'1'}     | ${'1'}      | ${true}
+    ${'1'}     | ${'0'}      | ${false}
+    ${'1.2.3'} | ${'1'}      | ${true}
+    ${'1.2'}   | ${'1'}      | ${true}
+    ${'1.0.0'} | ${'1'}      | ${true}
+    ${'1.2.0'} | ${'1.2'}    | ${true}
+    ${'1.2.3'} | ${'1.2'}    | ${true}
+    ${'0'}     | ${'latest'} | ${true}
+  `(
+    'matches("$version", "$range") === $expected',
```

Comment:
Some of the above are not valid versions, so these tests don't seem valid

Reply:
Should matches return false if the version is invalid, or should I just remove the tests which are using invalid versions?

Would updating the _compare method to return 1 when the version passed is invalid be the correct approach?

Reply:
In theory it should be a valid version passed to matches, but you can add a safety `if (!isVersion(Version)) { return false }` to `matches()`

Reply:
Ok I have added that check now